src/functionaltests/java/com/ericsson/ei/subscriptions/authentication/AuthenticationSteps.java

```diff
@@ -2,104 +2,110 @@
 
 import com.ericsson.ei.controller.model.GetSubscriptionResponse;
 import com.ericsson.ei.utils.FunctionalTestBase;
+import com.ericsson.ei.utils.HttpGetRequest;
+import com.ericsson.ei.utils.HttpPostRequest;
+import com.ericsson.ei.utils.HttpRequest;
 import com.ericsson.ei.utils.TestLDAPInitializer;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
+import cucumber.api.java.Before;
 import cucumber.api.java.en.Given;
 import cucumber.api.java.en.Then;
 import cucumber.api.java.en.When;
 import org.apache.commons.io.FileUtils;
 import org.apache.tomcat.util.codec.binary.Base64;
-import org.apache.tomcat.util.codec.binary.StringUtils;
 import org.json.JSONObject;
 import org.junit.Ignore;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
-import org.springframework.http.HttpHeaders;
+import org.springframework.boot.web.server.LocalServerPort;
 import org.springframework.http.HttpStatus;
-import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.web.servlet.MockMvc;
-import org.springframework.test.web.servlet.MvcResult;
-import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
 
 import java.io.File;
 
 import static org.junit.Assert.assertEquals;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 @Ignore
-@AutoConfigureMockMvc
 @ContextConfiguration(initializers = TestLDAPInitializer.class)
 public class AuthenticationSteps extends FunctionalTestBase {
 
     private static final String SUBSCRIPTION = "src/functionaltests/resources/subscription_single.json";
-
-    @Autowired
-    private MockMvc mockMvc;
-
-    private MvcResult mvcResult;
-
-    private String requestBody;
+    private static final String SUBSCRIPTION_NAME = "Subscription_Test";
+
+    @LocalServerPort
+    private int applicationPort;
+    private HttpRequest httpRequest;
+    private ResponseEntity<String> response;
+
+    @Before("@RESTWithCredentials,@RESTWithSession")
+    public void beforeScenario() throws Throwable {
+        httpRequest = new HttpGetRequest();
+        httpRequest.setUrl("http://localhost:").setPort(applicationPort).setEndpoint("/auth/logout");
```

Comment:
I don't remember if we decided to reuse functional tests in Systemtest level.
If its going to be used in system test level,, hostname need to be fetched from configuration.
On Dockerhost/Kubernetes each component will have different hostname alias.

Reply:
This comment applies for all hardcoded localhosts occurrences.

Reply:
right now we do not know how the configuration will be passed but it is good to implement a method that returns the host and then we can modify that method later to fetch host from property